src/site/lab10.rst

```diff
@@ -78,6 +78,25 @@ Bubble Up
 Bubble Down
 -----------
 
+* Given the min heap's property of the smallest element being at the root, I may be using this data structure for keeping track of ordered data
+    * I can easily access the minimum value from the collection since it is always at :math:`i = 0`
+
+* Unfortunately, if the minimum value is to be *removed*, there will be no value at the root of the tree
+    * A solution to this problem is to *bubble down*
+
+* Remove the element at the last index in the heap and place it in the root position :math:`i = 0`
+* Compare the moved value with its left and right children at indices :math:`2i + 1` and :math:`2i + 2`
+    * Swap the value with the smaller of the two child's value
+
+* Repeat this process until either
+    * The value is not greater than either children
+    * There are no more children to compare to; the value is at a leaf
+
+* It is important to always swap with the smaller of the two children when bubbling down since it ensures the min heap property
+    * If the value at the root of the (sub)tree :math:`x` is less than both the left and right child's values :math:`y` and :math:`z`
+    * And :math:`y < z`
+    * Then :math:`y` must become the root of the (sub)tree, otherwise :math:`z` would be a parent to :math:`y`, but :math:`z > y`
```

Comment:
This is a little wordy, but I guess it's more for the supernerds?

Reply:
🔪